submissions/svart0222/port-sniffer/index.js

```diff
@@ -0,0 +1,75 @@
+const net = require('net');
+
+const checkPort = (port, host) => {
+  return new Promise(resolve => {
+    const socket = net.createConnection(port, host, () => {
+      clearTimeout(timer);
+      resolve({ port, result: true });
+      socket.end();
+    });
+    const timer = setTimeout(() => {
+      resolve({ port, result: false });
+      socket.end();
+    }, 300);
+    socket.on('error', () => {
+      clearTimeout(timer);
+      resolve({ port, result: false });
+    });
+  });
+};
+
+const getResult = async (host, startPoint = 0, endPoint = 65535) => {
+  try {
+    const result = [];
+    let currentPoint = startPoint;
+
+    while (currentPoint < endPoint) {
+      result.push(checkPort(currentPoint, host));
+      currentPoint += 1;
+    }
+    const scanResult = await Promise.all(result);
```

Comment:
Look at comment bellow.
This looks like:
1. [....a lot of promises can be here]
2. lest resolve everything there

What will happen, if the number of operations in your call stack will be > 65,3k?

That's why, if you are using iterations approach, let's do async calls like they are sync.